sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -127,10 +136,53 @@ def main():
         sbom = json.load(f)
 
     base_images_sbom_components = get_base_images_sbom_components(base_images_digests, is_last_from_scratch)
-    if "formulation" in sbom:
-        sbom["formulation"].append({"components": base_images_sbom_components})
+    if args.sbom_type == "cyclonedx":
+        if "formulation" in sbom:
+            sbom["formulation"].append({"components": base_images_sbom_components})
+        else:
+            sbom.update({"formulation": [{"components": base_images_sbom_components}]})
     else:
-        sbom.update({"formulation": [{"components": base_images_sbom_components}]})
+        packages = []
+        relationships = []
+        for component in base_images_sbom_components:
+            SPDXID = (
+                f"SPDXRef-{component['type']}-{component['name']}-"
+                + f"{hashlib.sha256(component['purl'].encode()).hexdigest()}"
+            )
```

Comment:
What is the use case for the SPDXID?
- Is this something that will have to stay forever backwards compatible, because the IDs may be used as cross-SBOM references?
- Is there some reasoning behind the way the value is generated? Is it a good fit for the use cases?

Reply:
It's unique identifier which can be used externally, so yes, it's something permanent. 

> Is there some reasoning behind the way the value is generated? Is it a good fit for the use cases?
syft generates it in similar way, but it uses some go hasher which calculates hash of whole structure

Reply:
> It's unique identifier which can be used externally, so yes, it's something permanent.

Ack, please add a comment about that

> syft generates it in similar way, but it uses some go hasher which calculates hash of whole structure

Hmm, interesting. If two elements have the same purl but some other property is different, should they be considered the same element? I guess Syft's answer is No, but maybe it should be Yes.

Reply:
yeah, I would say syft would generate different SPDXID, but question is if there isn't some deduplication mechanism which it uses internally.

Reply:
Might be, but this also applies cross-SBOM. If two different SBOMs have a component with the same purl, should it have the same SPDXID? Probably yes.

Reply:
I think if you refer to package in another document it has to be referenced by the spdxid of the document and its own spdxid.
"
7.2 Package SPDX identifier field
7.2.1 Description

Uniquely identify any element in an SPDX document which may be referenced by other elements. These may be referenced internally and externally with the addition of the SPDX document identifier.
"

Reply:
Ack, makes sense.